Fix nil error

niklas opened the pull request:
The Form builder provides access to the object of the form, we should use it. Additionally, the type check for the translatable field should be done on the `Translation`, not on the original model.

niklas committed:
Use the Translation itself to determine its type

the original @resource might not actually have the original field
(anymore)

niklas committed:
Avoid ivars in partial, ask form builder instead

niklas commented:
this fixes #17

tvdeyen commented:
Thanks @niklas. Could you please have a look into the failing spec?

niklas commented:
I did. Here are my results: 

1) `solidus_globalize` wants to translate `Spree::ProductProperty#value` (beside others)
2) `globalize` sets everything up, complains about the `value` column still existing
```
DEPRECATION WARNING: You have defined translated attributes with names that conflict with columns on the model table. Globalize does not support this configuration anymore, remove or rename these columns.
Model name (table name): Spree::ProductProperty (spree_product_properties)
Attribute name(s): value
```

3) `globalize` adds `"value"` to `Spree::ProductProperty.ignored_columns`
(is considered a fix https://github.com/globalize/globalize/pull/560)

4) a custom validator in `solidus` breaks because the column is not "present" anymore (it is, but not visible to any AR introspection), but is assumed it is.

I currently have no idea how to solve this particular problem reliably. We could patch the Validator to avoid the `NilError`, but `globalize` wants us to remove the translated columns from the original tables entirely. This might bring update pains later on when `solidus` decides to `ALTER COLUMN` or whatever.

taniarv commented:
I am having the same issue and I have fixed it creating a spree_product_property decorator
app/models/spree/product_property_decorator.rb

```
module Spree
  ProductProperty.class_eval do
    # remove all validations
    _validate_callbacks.clear
    # define property presence validator 
    validates :property, presence: true
  end
end
```

niklas committed:
Remove Spree Validation depending on a column hidden by globalize.

niklas committed:
Make Product finable by slug again.

globalize hides the slug column, which causes find_by_slug to not be
available for use.

tvdeyen commented:
There is a build failure: the migrations do not run because of the removed `:slug` column. This needs to be dealt with. 

I'm closing this for now as this is very outdated and needs to be opened again after the issues are resolved. 

Thanks for your contribution!
